stl/inc/xloctime

```diff
@@ -132,6 +132,9 @@ public:
                 }
 
                 _First = do_get(_First, _Last, _Iosbase, _State, _Pt, _Specifier, _Modifier); // convert a single field
+                if (_State != 0) {

```

Comment:
can we get a comment on why this is here?

Is this to handle the case where some derived class recognizes a field and the base class doesn't?

Reply:
or is it another bit of git munging?

Reply:
> or is it another bit of git munging?

I added these three lines to fix the bug I described [Here](https://github.com/microsoft/STL/issues/1126#issuecomment-670076039)
for example, imagine we are trying to parse this string "1990-M-25". Without these three lines, the year will parse successfully so `_State` would be 0. On the next iteration, we try to parse month that will fail and _State would be non zero. then we continue to parse Day, and the _State will be overwritten by 0, so the final _State will consider as a successful parse but month field will be untouched with its default value which is 0 (i.e, January)